drivers/sensor/sensor_shell.c

```diff
@@ -360,12 +362,10 @@ void sensor_shell_processing_callback(int result, uint8_t *buf, uint32_t buf_len
 			continue;
 		}
 
-		while (decoder->get_frame_count(buf, channel, channel_idx, &frame_count) == 0) {
```

Comment:
Since we're no longer using `channel_idx`, let's then increase `chan_spec.chan_idx` on every loop iteration (Otherwise, issuing `sensor get <sensor-id>` results in an endless loop of `chan_idx = 0` readings). The suggested fix:

``` c

	for (int channel = 0; channel < SENSOR_CHAN_ALL; ++channel) {
		...
		while (decoder->get_frame_count(buf, chan_spec, &frame_count) == 0) {
			...
			++chan_spec.chan_idx;
		}

```

I'd even go further and suggest scrubbing this local variable entirely as it's no longer needed.

Reply:
Suggestion brought up on the Sensor WG Meeting: We probably want to add a testsuite verifying the sensor shell commands are kept functional across changes, in a generalized fashion, as the generic test covers all sensors.

(https://github.com/zephyrproject-rtos/zephyr/pull/72052 adds a test-suite for the proximity sensor shell commands)

In any case, not considered in scope for this PR. Just capturing it so we don't forget.